libbeat/esleg/eslegclient/bulkapi.go

```diff
@@ -71,14 +71,16 @@ func (conn *Connection) Bulk(
 		return 0, nil, nil
 	}
 
+	mergedParams := mergeParams(conn.ConnectionSettings.Parameters, params)
```

Comment:
Nit: I would merge the params more closely before they are used. If there is an error in the bulkEncode we never would have to merge them. Same below.